dotnet/src/SemanticKernel.Abstractions/Memory/IMemoryRecordMapper.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Memory;
+
+/// <summary>
+/// Interface for mapping between a storage model, and the consumer data model.
+/// </summary>
+/// <typeparam name="TConsumerDataModel">The consumer data model to map to or from.</typeparam>
+/// <typeparam name="TStorageModel">The storage model to map to or from.</typeparam>
+public interface IMemoryRecordMapper<TConsumerDataModel, TStorageModel>
```

Comment:
nit: If we will use `VectorStore` naming, we should probably use `TStoreModel` here as well.

Reply:
Thanks, I'll update when I do the rest of the renaming.